qiime/sdk/tests/test_method.py

```diff
@@ -0,0 +1,379 @@
+            ])
+        )
+        self.assertEqual(method.signature, exp_sig)
+
+        self.assertEqual(method.name, 'Merge mappings')
+        self.assertTrue(
+            method.description.startswith('This method merges two mappings'))
+        self.assertTrue(
+            method.source.startswith('\n```python\ndef merge_mappings('))
+
+    def test_is_callable(self):
+        self.assertTrue(callable(self.plugin.methods['concatenate_ints']))
+        self.assertTrue(
+            callable(self.plugin.methods['concatenate_ints_markdown']))
+
+    def test_callable_properties(self):
+        concatenate_ints = self.plugin.methods['concatenate_ints']
+        concatenate_ints_markdown = self.plugin.methods['concatenate_ints']
+        merge_mappings = self.plugin.methods['merge_mappings']
+
+        for method in (concatenate_ints, concatenate_ints_markdown,
+                       merge_mappings):
+            self.assertEqual(method.__call__.__name__, '__call__')
+            self.assertEqual(method.__call__.__annotations__, {})
+            self.assertFalse(hasattr(method.__call__, '__wrapped__'))
+
+    def test_async_properties(self):
+        concatenate_ints = self.plugin.methods['concatenate_ints']
+        concatenate_ints_markdown = self.plugin.methods['concatenate_ints']
+        merge_mappings = self.plugin.methods['merge_mappings']
+
+        for method in (concatenate_ints, concatenate_ints_markdown,
+                       merge_mappings):
+            self.assertEqual(method.async.__name__, 'async')
+            self.assertEqual(method.async.__annotations__, {})
+            self.assertFalse(hasattr(method.async, '__wrapped__'))
+
+    def test_callable_and_async_signature(self):
+        concatenate_ints = self.plugin.methods['concatenate_ints']
+        concatenate_ints_markdown = self.plugin.methods['concatenate_ints']
+
+        for method in concatenate_ints, concatenate_ints_markdown:
+            for callable_attr in '__call__', 'async':
+                signature = inspect.Signature.from_callable(
+                    getattr(method, callable_attr))
+                parameters = list(signature.parameters.items())
+
+                kind = inspect.Parameter.POSITIONAL_OR_KEYWORD
+                exp_parameters = [
+                    ('ints1', inspect.Parameter('ints1', kind)),
+                    ('ints2', inspect.Parameter('ints2', kind)),
+                    ('ints3', inspect.Parameter('ints3', kind)),
+                    ('int1', inspect.Parameter('int1', kind)),
+                    ('int2', inspect.Parameter('int2', kind))
+                ]
+                self.assertEqual(parameters, exp_parameters)
+
+                self.assertEqual(signature.return_annotation,
+                                 inspect.Signature.empty)
+
+    def test_callable_and_async_different_signature(self):
+        # Test that a different Method object has a different dynamic
+        # signature.
```

Comment:
Can you provide the test dependency `test_callable_and_async_signature` in this comment and vice-versa? I was super confused for a few minutes how it was being tested. This test only works in the context of the other existing. Alternatively (to avoid any potential trickery by `setUp`) we could just test two different methods here (a little redundant, but then the test has no external dependencies for its claim)